src/wallet-redux.js

```diff
@@ -186,26 +182,9 @@ export function reducer(state, action) {
 
         case TypeKeys.REMOVE_ACCOUNT: {
             const accounts = new Map(state.accounts);
-            accounts.delete(action.address);
 
-            const accountArray = [...accounts.values()];
-
-            const affectedVestingContracts = accountArray.filter(account =>
-                account.type === AccountType.VESTING && account.owner === action.address
-            );
-
-            const affectedHTLCContracts = accountArray.filter(account =>
-                account.type === AccountType.HTLC
-                && !accounts.get(account.sender)
-                && !accounts.get(account.recipient)
-            );
-
-            for (const contract of affectedVestingContracts) {
-                accounts.delete(contract.address);
-            }
-
-            for (const contract of affectedHTLCContracts) {
-                accounts.delete(contract.address);
+            for (const address of action.addressesToKeep) {
+                accounts.delete(address);
```

Comment:
Haha, what? Now you delete all addresses that should be kept! :astonished:

Reply:
Test passed :p

Reply:
Because `REMOVE_ACCOUNT` is currently unused. Or did you test it after changing this? :stuck_out_tongue:

Reply:
I mean, YOU passed my test. ^^

Reply:
(It's in use because it's now the same as logout_legacy.)